templates/python/my_first_blueprint.py

```
@@ -159,13 +159,14 @@ def my_first_validation(self, target_resource: KMS, resource_metadata: ResourceM
     # ---------------------------------------------------------------------
     # Validation Function Caller
     # ---------------------------------------------------------------------
-    def validate(self) -> List[Finding]:
+    def validate(self) -> List[Finding | Exception]:
```

Potentially controversial change.

I see blueprints as unit tests, and unit tests can fail. But because we don't control how the tests are run, and we just call this `validate()` function that could potentially call a bunch of other "smaller" functions, there has to be a way to collect errors. Some of those granular functions may return Findings, and some may fail.

Which is why I expanded the return type to be either Findings or exceptions.

---

Ideally I think tython would call the granular functions, and then it can try catch them to collect the exceptions. But since we're delegating that responsibility to the user, it seems like we have to also give the user the ability to collect the exceptions...

---

How exceptions are handled upstream I think is irrelevant. We shouldn't silently ignoring the exceptions that's for sure, so they have to be captured, but then what we do to them and how we tell the user about those exceptions is a different problem.

---

Removing the controversial change for now. Don't wanna block the rest the PR because of it. We'll revisit later.